### DISCUSSION TITLE
Unknown field `_count` for include statement...

### DISCUSSION THREAD

**Discussion by alucardu** [prisma/prisma#9505] [Q&A] [answered]
I have a function that returns a list of `Notifications` related to the `User`:

```ts
const returnUserNotifications = async (args, req) => {
  const user = await prisma.user.findUnique({
    where: {id: req.userId},
    include: {
      notifications: {
        orderBy: {
          id: 'desc',
        },
        include: {
          movie: true,
          followedUser: true,
        },
      },
    },
  });
  return user.notifications;
};
```

I would like to add a a count of the amount of non **watched** `Notifications`.  A `Notification` has a boolean value **watched** which is false on initialization but gets set to true if a user clicks on the notification.

The docs show:

```ts
const usersWithCount = await prisma.user.findMany({
  include: {
    _count: {
      select: { posts: true },
    },
  },
})
```

But when I add my `_count` I get an error:

```

Invalid `prisma.user.findUnique()` invocation:

{
  where: {
    id: 2
  },
  include: {
    _count: {
    ~~~~~~
      select: {
        notifications: true
      }
    },
?   notifications?: true,
?   movies?: true,
?   followedBy?: true,
?   following?: true,
?   Notification?: true
  }
}


Unknown field `_count` for include statement on model User. Available options are listed in green. Did you mean `movies`?
```

Also I think I can't filter this count, so it will only return a count of all `Notifications`. Not only `Notifications` which the **wached** field of false. 

So I think I need an additional query to get the [count and filter](https://www.prisma.io/docs/concepts/components/prisma-client/aggregation-grouping-summarizing#filtered-count) like so:

```
  const notificationCount = await prisma.notification.count({
    where: {
      userId: req.userId,
      watched: false,
    },
  });
```

But I would love to do this in the `returnUserNotifications` query.

**Comment by ryands17** [comment]
@alucardu 👋 

What version of Prisma are you on?

**Reply by alucardu** [reply]
Sorry should've posted. I'm on `"prisma": "2.30.0",`.

**Reply by ryands17** [reply, marked as answer]
As you're on the older version, you need to add `selectRelationCount` to the `previewFeatures` to the `generator` block in your `schema.prisma` and then run `prisma generate`. Otherwise, update `prisma` and `@prisma/client` to the latest version.

**Reply by alucardu** [reply]
Kind of odd that the docs say:

> Count relations. The ability to count relations is available in version 2.20.0 and later.

**Reply by ryands17** [reply]
Available as a preview feature in 2.20.0, but made generally available in 3.0.0

**Reply by ryands17** [reply]
We will add a note there in the docs for this.

**Reply by alucardu** [reply]
Is it possible to filter using the `count` API?

```
_count: {
  select: {
    notifications: true,
  },
},
```

Returns the count of all the notifications. I want to filter that count on a `watched` boolean field inside the `notification` model.  Something like:

```
const unwatchedNotifications = await prisma.notification.count({
  where: {
    userId: req.userId,
    watched: false,
  },
});
```

**Reply by ryands17** [reply]
Unfortunately not. Filtering relation counts is not supported at the moment. A 👍 on this request would be great so that we know the priority for this 🙂 
https://github.com/prisma/prisma/issues/8413

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
